python/lsst/meas/algorithms/brightStarStamps.py

```diff
@@ -0,0 +1,280 @@
+        ``"gaiaGMag"``
+            G magnitude.(`float`).
+        ``"gaiaId"``
+            Unique Gaia identifier (`str`).
+        `"annularFlux"`
+            Bright star normalization factor, computed by integration over an
+            annulus (`float`)
+    innerRadius, outerRadius : `float`, optional
+        Radii values, in pixels, that define the annulus used to compute the
+        ``"annularFlux"`` values within each ``starStamp``. Must be provided
+        if ``"INNER_RADIUS"`` and ``"OUTER_RADIUS"`` are not present in
+        ``metadata``.
+    metadata : `lsst.daf.base.PropertyList`, optional
+        Metadata associated with the bright stars.
+
+    Raises
+    ------
+    ValueError
+        Raised if one of the star stamps provided does not contain the
+        required keys.
+    AttributeError
+        Raised if the definition of the annulus used to compute each star's
+        normalization factor are not provided, that is, if ``"INNER_RADIUS"``
+        and ``"OUTER_RADIUS"`` are not present in ``metadata`` _and_
+        ``innerRadius`` and ``outerRadius`` are not provided.
+
+    Examples
+    --------
+    Read full bright star stamps using the butler:
+
+    >>> starStamps = butler.get("brightStarStamps", dataId)
+
+    A (gen2) butler can also be used to read only a part of the stamps,
+    specified by a bbox:
+
+    >>> starSubregions = butler.get("brightStarStamps_sub", dataId, bbox=bbox)
+    """
+
+    def __init__(self, starStamps, innerRadius=None, outerRadius=None,
+                 metadata=None,):
+        self._starStamps = [self._checkStarStamp(stamp) for stamp in starStamps]
+        self._setMetadata(metadata)
+        # Add inner and outer radii to metadata
+        self._setAnnulusRadii(innerRadius, outerRadius)
+
+    def __len__(self):
+        return len(self._starStamps)
+
+    def __getitem__(self, index):
+        return self._starStamps[index]
+
+    def append(self, item):
+        """Add an additional bright star stamp.
+
+        Parameters
+        ----------
+        item : `dict`
+            Mapping containing keys ``"starStamp"``, ``"gaiaGMag"``,
+            ``"gaiaId"``, and `"annularFlux"`.
+        """
+        self._starStamps.append(self._checkStarStamp(item))
+        return None
+
+    def getMaskedImages(self):
+        """Retrieve star images.
+
+        Returns
+        -------
+        maskedImages :
+            `list` [`lsst.afw.image.maskedImage.maskedImage.MaskedImageF`]
+        """
+        return [stamp["starStamp"] for stamp in self._starStamps]
+
+    def getMagnitudes(self):
+        """Retrieve Gaia G magnitudes for each star.
+
+        Returns
+        -------
+        gaiaGMags : `list` [`float`[
+        """
+        return [stamp["gaiaGMag"] for stamp in self._starStamps]
+
+    def getGaiaIds(self):
+        """Retrieve Gaia IDs for each star.
+
+        Returns
+        -------
+        gaiaIds : `list` [`str`]
```

Comment:
Can I ask why str here?

Reply:
I think I figured it made sense for unique ids to be strings, even if (as seems to be the case here with Gaia DR2) they happen to be ints. Now that I actually take a step back and think about it, I cannot really justify it, I suppose.

Would you recommend specifying int, or not providing the type at all?